### PR TITLE
Enables fetching scala library sources

### PR DESCRIPTION
**Pull request by nmarton-da**
Enables fetch_sources flag for fetching scala sources.
These sources can be used for manual linking scala library if sources are not resolved otherwise.
Motivation: https://github.com/bazelbuild/rules_scala/issues/908

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Looks very sensible, thank you @nmarton-da!

**Comment by cocreature** [comment]
Looks like you need to reformat, run `./fmt.sh` or `./fmt.sh --diff` if you need something faster.

**Comment by nmarton-da** [comment]
> Looks like you need to reformat, run `./fmt.sh` or `./fmt.sh --diff` if you need something faster.

yes, just figured, ran that :+1:

**Review by SamirTalwar-DA — APPROVED** [review]

**Comment by SamirTalwar-DA** [comment]
If this doesn't work automatically, can you add some information on how to get it working in _BAZEL-JVM.md_?

File: BAZEL.md [hunk 1]
```diff
@@ -300,6 +300,14 @@ Bazel command (`run`, `test`, etc.). If applicable, you can also define Bazel
 command-line flags or command-line flags to the executable. Click on "Apply",
 or "OK" to add the run configuration.
 
+#### Attaching sources to scala library
+
+If you do not have the scala library sources linked (you only see the decompiled
+ sources only), you can attach it manually by selecting the `Choose sources...` 
+ button on the yellow bar at the top, and selecting `scala-library...-src.jar`. 
+(you might need to update dependencies beforehand by running 
+`bazel run @unpinned_maven//:pin`)
```
Comment (SamirTalwar-DA) [inline BAZEL.md:309]: You shouldn't need to do this unless you change _bazel-java-deps.bzl_. What does it achieve?

I would have thought that asking the user to _Sync Project with BUILD Files_ would be enough.

File: BAZEL.md [hunk 2]
```diff
@@ -300,6 +300,14 @@ Bazel command (`run`, `test`, etc.). If applicable, you can also define Bazel
 command-line flags or command-line flags to the executable. Click on "Apply",
 or "OK" to add the run configuration.
 
+#### Attaching sources to scala library
+
+If you do not have the scala library sources linked (you only see the decompiled
```
Comment (SamirTalwar-DA) [inline BAZEL.md:305]:
```suggestion
If you do not have the Scala library sources linked (you only see the decompiled
```

File: BAZEL.md [hunk 3]
```diff
@@ -300,6 +300,14 @@ Bazel command (`run`, `test`, etc.). If applicable, you can also define Bazel
...
+ sources only), you can attach it manually by selecting the `Choose sources...`
```
Comment (SamirTalwar-DA) [inline BAZEL.md:306]:
```suggestion
 sources), you can attach it manually by selecting the `Choose sources...` 
```

File: BAZEL.md (hunk 1 above)
Comment (nmarton-da) [reply BAZEL.md]: Sounds great! removing it

**Pull request merged**